challenge_storing_groceries/src/challenge_storing_groceries/pdf.py

```diff
@@ -46,10 +49,16 @@ def execute(self, userdata=None):
         # ToDo: store probabilities in the world model
 
         # Get DETECTED_OBJECTS_WITH_PROBS, i.e., the detections resulting from inspection
-        for entity, probability in config.DETECTED_OBJECTS_WITH_PROBS:
+        for entity, probability in self.initial_inspection_ds.DETECTED_OBJECTS_WITH_PROBS:
             if entity.id not in self._items:
                 image = save_entity_image_to_file(self._robot.ed, entity.id)
-                self._items[entity.id] = (entity, probability, image)
+                self._items[entity.id] = (entity, probability, image,"initialinspection",len(self._items))
```

Comment:
Why do you need to store the length of this dict (before adding?)?

Reply:
Maybe, another data structure than a tuple is more suitable. You could, e.g., create a new class to this end or create a so-called namedtuple

Reply:
Instead of "initialinspection" (and later on "fromtable") it might be nice to use the id of the entity you're inspecting.